core/schema/host.go

```diff
@@ -413,17 +414,29 @@ func loadDirectoryGitIgnorePatterns(ctx context.Context, parentPath string, host
 				{Name: "noGitAutoIgnore", Value: dagql.Boolean(true)},
 			},
 		},
+		dagql.Selector{
+			Field: "__gitIgnoreFor",
+		},
 	)
 	if err != nil {
 		return nil, fmt.Errorf("failed to load directory ignore patterns: %w", err)
 	}
 
-	gitIgnorePatterns, err := core.LoadGitIgnoreInDirectory(ctx, directory)
-	if err != nil {
-		return nil, fmt.Errorf("failed to load .gitignore in context directory %q: %w", parentPath, err)
+	for i := 1; i <= result.Len(); i++ {
+		entry, err := result.Nth(i)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get entry %d in directory ignore patterns: %w", i, err)
+		}
+
+		entryValue, ok := dagql.UnwrapAs[dagql.String](entry)
```

Comment:
We don't need `UnwrapAs`, you can just call `string(entry)`, since it's just a string type.

Reply:
Oh. Just call `Self`.